spot_wrapper/spot_arm.py

```diff
@@ -0,0 +1,561 @@
+
+                # Command issue with RobotCommandClient
+                self._robot_command_client.robot_command(unstow)
+                self._logger.info("Command unstow issued")
+                time.sleep(2.0)
+
+        except Exception as e:
+            return False, "Exception occured while trying to unstow"
+
+        return True, "Unstow arm success"
+
+    def arm_carry(self) -> typing.Tuple[bool, str]:
+        try:
+            success, msg = self.ensure_arm_power_and_stand()
+            if not success:
+                self._logger.info(msg)
+                return False, msg
+            else:
+                # Get Arm in carry mode
+                carry = RobotCommandBuilder.arm_carry_command()
+
+                # Command issue with RobotCommandClient
+                self._robot_command_client.robot_command(carry)
+                self._logger.info("Command carry issued")
+                time.sleep(2.0)
+
+        except Exception as e:
+            return False, "Exception occured while carry mode was issued"
+
+        return True, "Carry mode success"
+
+    def make_arm_trajectory_command(
+        self, arm_joint_trajectory: arm_command_pb2.ArmJointTrajectory
+    ) -> robot_command_pb2.RobotCommand:
+        """Helper function to create a RobotCommand from an ArmJointTrajectory.
+        Copy from 'spot-sdk/python/examples/arm_joint_move/arm_joint_move.py'"""
+
+        joint_move_command = arm_command_pb2.ArmJointMoveCommand.Request(
+            trajectory=arm_joint_trajectory
+        )
+        arm_command = arm_command_pb2.ArmCommand.Request(
+            arm_joint_move_command=joint_move_command
+        )
+        sync_arm = synchronized_command_pb2.SynchronizedCommand.Request(
+            arm_command=arm_command
+        )
+        arm_sync_robot_cmd = robot_command_pb2.RobotCommand(
+            synchronized_command=sync_arm
+        )
+        return RobotCommandBuilder.build_synchro_command(arm_sync_robot_cmd)
+
+    def arm_joint_move(self, joint_targets) -> typing.Tuple[bool, str]:
+        # All perspectives are given when looking at the robot from behind after the unstow service is called
+        # Joint1: 0.0 arm points to the front. positive: turn left, negative: turn right)
+        # RANGE: -3.14 -> 3.14
+        # Joint2: 0.0 arm points to the front. positive: move down, negative move up
+        # RANGE: 0.4 -> -3.13 (
+        # Joint3: 0.0 arm straight. moves the arm down
+        # RANGE: 0.0 -> 3.1415
+        # Joint4: 0.0 middle position. negative: moves ccw, positive moves cw
+        # RANGE: -2.79253 -> 2.79253
+        # # Joint5: 0.0 gripper points to the front. positive moves the gripper down
+        # RANGE: -1.8326 -> 1.8326
+        # Joint6: 0.0 Gripper is not rolled, positive is ccw
+        # RANGE: -2.87 -> 2.87
+        # Values after unstow are: [0.0, -0.9, 1.8, 0.0, -0.9, 0.0]
+        if abs(joint_targets[0]) > 3.14:
```

Comment:
All of these values should be loaded from a configuration file or the URDF itself.

Reply:
I don't have any experience using the arm, but are these joint limit checks even necessary? Based on other commands, the motion should be limited on the robot anyway. 

If we do need them, I think the way to access is to use [`get_robot_hardware_configuration`](https://dev.bostondynamics.com/python/bosdyn-client/src/bosdyn/client/robot_state.html#bosdyn.client.robot_state.RobotStateClient.get_robot_hardware_configuration) to retrieve the robot skeleton, which returns [`RobotHardwareConfigurationResponse`](https://dev.bostondynamics.com/protos/bosdyn/api/proto_reference.html?highlight=urdf#robothardwareconfigurationresponse). Probably the best way to access the URDF is to use `urdfpy`, which seems to allow easy access to joint limits (actually it seems that library is no longer maintained, perhaps https://github.com/fishbotics/urchin/ would be better to use). It can only load URDF from file, so would need to save to a tmp file first